docs/contributing/Building, Debugging, and Testing on Windows.md

```diff
@@ -0,0 +1,91 @@
+# Required Software
+
+1. [Visual Studio 2015 with Update 1](http://go.microsoft.com/fwlink/?LinkId=691129). _You need Update 1_.
+2. Visual Studio 2015 Extensibility Tools. If you already installed Visual Studio, choose "Modify" from the Programs and Features control panel, and check "Visual Studio Extensibility".
+
+# Getting the Code
+
+1. Clone https://github.com/dotnet/roslyn
+2. Run the "Developer Command Prompt for VS2015" from your start menu.
+3. Run `Restore.cmd` in the command prompt to restore NuGet packages.
+4. Due to [Issue #5876](https://github.com/dotnet/roslyn/issues/5876), you should build on the command line before opening in Visual Studio. Run `msbuild /v:m /m Roslyn.sln`
+5. Open _Roslyn.sln_
+
+# Running Tests
+
+Tests cannot be run via Test Explorer due to some Visual Studio limitations.
+
+1. Run the "Developer Command Prompt for VS2015" from your start menu.
+2. Run `msbuild /v:m /m BuildAndTest.proj` in the command prompt.
+
+To debug through tests, you can right click the test project that contains your
+tests and choose **Set as Startup Project**. Then press F5. This will run the
+tests under the command line runner.  Some members of the team have been
+working on a GUI runner that allows selection of individual tests, etc.  Grab
+the source from
+[xunit.runner.wpf](https://github.com/pilchie/xunit.runner.wpf), build it and
+give it a try.
+
+# Trying Your Changes in Visual Studio
+
+Starting with Update 1, it is now possible to run your changes inside Visual
+Studio to try them out. Some projects in Roslyn.sln, listed below, build Visual
+Studio extensions. When you build those projects, they automatically deploy
+into an experimental instance of Visual Studio. The first time you clone, you
+should first do a full build of Roslyn.sln to make sure everything is primed.
+Then, you can run Visual Studio by right clicking the appropriate project in
+Visual Studio, setting it as a startup project, and pressing F5. You can also
+run Visual Studio after building by running a "Developer Command Prompt for
+VS2015" and then running `devenv /rootsuffix RoslynDev`.
+
+Here are what is deployed with each extension, by project that builds it. If
+you're working on a particular area, you probably want to set the appropriate
+project as your startup project to ensure the right things are built and
+deployed.
+
+- **VisualStudioSetup**: this project builds Roslyn.VisualStudio.Setup.vsix. It
```

Comment:
It does not. That's why I'm hedging with "you should do a full build to make sure everything is primed." I'm undecided whether this is good or bad -- it means we're only redeploying on what you want, but at the same time it might cause problems.